src/client.ts

```diff
@@ -0,0 +1,93 @@
+import { PoolConfig } from 'pg';
+import { PostgresSystemDatabase, SystemDatabase, WorkflowStatusInternal } from './system_database';
+import { GlobalLogger as Logger } from './telemetry/logs';
+import { v4 as uuidv4 } from 'uuid';
+import { RetrievedHandle, StatusString, WorkflowHandle } from './workflow';
+
+interface EnqueueOptions {
+  queueName: string;
+  workflowName: string;
+  workflowClassName: string;
+  workflowUUID?: string;
+  maxRetries?: number;
+  appVersion?: string;
+}
+
+export class DBOSClient {
+  readonly logger: Logger;
+
+  readonly systemDatabase: SystemDatabase;
+
+  constructor(poolConfig: PoolConfig, systemDatabase: string) {
```

Comment:
I wonder if it's better to pass in a single database url to their system database.

Reply:
As we're moving towards a connection-based config, we should take in a connection string and (optionally) a system database name.  This requires some code for https://github.com/dbos-inc/dbos-transact-ts/pull/800 to parse connection strings, so we need to hold off until that PR is merged to make the change.

Reply:
Agree 💯 will do after #800 is merged

Reply:
The configify PR is merged, let's modify this interface to take in a connection string and an optional system DB name.

Reply:
I know, I've been working on the "spawn a separate process and kill" stuff. I will update the signature to be a connection string before I merge